protos/sigstore_common.proto

```diff
@@ -140,6 +140,20 @@ message LogId {
         bytes key_id = 1 [(google.api.field_behavior) = REQUIRED];
 }
 
+// The checkpoint key ID, following the specification described here
+// for ECDSA and Ed25519 signatures:
+// https://github.com/C2SP/C2SP/blob/main/signed-note.md#signatures
+// For RSA signatures, the key ID will match the ECDSA format of the hashed
+// DER-encoded SPKI public key. Publicly witnessed logs MUST NOT use
+// RSA-signed checkpoints, since witnesses do not support RSA signatures.
+// This is provided for convenience. Clients can also calculate the checkpoint
+// key ID given the log's public key.
+message CheckpointKeyId {
+        // The key ID in a checkpoint, as a prefix to the signature. SHOULD be
+        // 4 bytes long, as a truncated hash.
+        bytes key_id = 1 [(google.api.field_behavior) = REQUIRED];
+}
```

Comment:
This is a minor thing, but this will mark at least the third overlapping type for this in the protobuf-specs 😅 

* `common.LogId`
* `common.PublicKeyIdentifier`
* (new) `common.CheckpointKeyId`

I have no objection to a new type in principle, but would either of the pre-existing ones work? `PublicKeyIdentifier` in particular might be ideal, since it isn't prescriptive in name like `LogId` and has an arbitrary `string hint`.

Reply:
`PublicKeyIdentifier` seems good. What's your preference on `string` vs `bytes`?

Reply:
I'd prefer `bytes` too. We could reuse `LogId`, but we'd need to move the comment on `key_id` up to the `TransparencyLogInstance` message. The motivation for a separate type would be to keep the spec comments alongside the message, but I think it can still be clean if moved out to a higher level. WDYT?

Reply:
> We could reuse LogId, but we'd need to move the comment on key_id up to the TransparencyLogInstance message. The motivation for a separate type would be to keep the spec comments alongside the message, but I think it can still be clean if moved out to a higher level. WDYT?

That seems reasonable to me!

Reply:
I actually like this structure, because we can now use `LogId` to house both `key_id` and a future `generic_log_id` or something like that if we move away from logs represented by their keys.